synapse/storage/events.py

```diff
@@ -591,10 +557,28 @@ def event_dict(event):
             ],
         )
 
-        if context.rejected:
-            self._store_rejections_txn(
-                txn, event.event_id, context.rejected
-            )
+        to_remove = set()
```

Comment:
ok I see it's deliberate. reusing the same variable is bad karma imho. But basically it stems from this method being too big. Factor some of this shit out.